examples/explorer/ui.rs

```diff
@@ -140,28 +140,25 @@ impl App {
                 .join("/");
             let price = update
                 .states
-                .get(&comp.address)
+                .get(id)
                 .map(|el| el.spot_price(&comp.tokens[0], &comp.tokens[1]))
                 .unwrap_or(Ok(0.0));
 
             self.items.push(Data {
                 component: comp.clone(),
-                state: update
-                    .states
-                    .get(&comp.address)
-                    .unwrap()
-                    .clone(),
+                state: update.states.get(id).unwrap().clone(),
                 name,
                 tokens,
                 price: format!("{}", price.expect("Expected f64 as spot price")),
             });
         }
 
         for (address, state) in update.states.iter() {
+            let eth_address = Bytes::from_str(address).expect("Bad address");
             let entry = self
                 .items
                 .iter()
-                .find_position(|e| e.component.address == *address);
+                .find_position(|e| e.component.address == eth_address);
```

Comment:
I would rename this to `ethereum_address`. `eth_address` seems it's the ETH address

Reply:
I think `ethereum_address` and `eth_address` have the same meaning in my head. For ETH address I would call it the `native_address` since that's what it's been throughout our defibot code - so in my opinion this is fine